### PR TITLE
Draft - Proposal for code linting

### PR DESCRIPTION
**Pull request by 1337LutZ**
Best to check out this branch and run the commands for yourself.

Didn't commit all the file changes that would happen because of this (over 300). 

Have a [branch ](https://github.com/1337LutZ/wowsims-wotlk/commit/349d64c0727f229a8b6b0b0f7d64728832b18fef) where I ran this command but haven't fixed all the code smells/warnings yet (there's a lot). If there's interest I can go through all the files and fixed them in one fell swoop.

- Added linting commands
- Added prettier & stylelint
- Extended eslint configuration
- Removed `.vscode` folder from `.gitignore` so project settings are picked up when the project is cloned
- Added `codeActionsOnSave` command for relevant files
- Added `.vscode/extensions.json` to notify developers for suggested extensions when working on this project

**commands**
- `npm run format` will auto fix and format css (scss) and js (ts) files
- `npm run lint` will lint all js and css files
- `npm run lint:{js/css}` will lint all js or css files
- `npm run lint:{js/css}:fix` will fix all js or css files based on linting rules
- `npm run prettier` will check all js, json and css files for formatting
- `npm run prettier:fix` will fix formatting for all js and css files

### CODE REVIEW

**Comment by 1337LutZ** [comment]
Hey @kayla-glick, Yea sure can do!
I would update this branch with the latest changes then. However once I run the linting there will be A LOT of changes (basically all the frontend). That's why I would prefer to first introduce the linting changes, and then you can validate this PR by running the lint commands yourself?

**Comment by 1337LutZ** [comment]
@kayla-glick I updated it with the latest master builds. Feel free to play around with it. I haven't changed any imports / warnings yet since it'll be over 200+ files.

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,17 @@
...
+WORKDIR /wotlk
+COPY . .
+COPY gitconfig /etc/gitconfig
+
+RUN apt-get update
+RUN apt-get install -y protobuf-compiler
+RUN go get -u google.golang.org/protobuf
+RUN go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
+
+ENV NODE_VERSION=18.18.0
```
Comment (kayla-glick) [inline Dockerfile:13]: I see this is downgraded from 19.X to 18.X. Was there a reason you'd need to do that?

Comment (1337LutZ) [reply Dockerfile:13]: No, not that I know off. I think this might be a faulty merge.

**Comment by 1337LutZ** [comment]
Sweet, nice to hear!
Just prepare for the 200+ files of changes when you run the linx fiz command 👌

**Pull request merged**